src/group_cmd.cpp

```diff
@@ -512,48 +512,63 @@ static void AddVehicleToGroup(Vehicle *v, GroupID new_g)
  * @param add_shared Add shared vehicles as well.
  * @return the cost of this operation or an error
  */
-std::tuple<CommandCost, GroupID> CmdAddVehicleGroup(DoCommandFlag flags, GroupID group_id, VehicleID veh_id, bool add_shared)
+std::tuple<CommandCost, GroupID> CmdAddVehicleGroup(DoCommandFlag flags, GroupID group_id, VehicleID veh_id, bool add_shared, const VehicleListIdentifier &vli)
 {
-	Vehicle *v = Vehicle::GetIfValid(veh_id);
 	GroupID new_g = group_id;
+	if (!Group::IsValidID(new_g) && !IsDefaultGroupID(new_g) && new_g != NEW_GROUP) return { CMD_ERROR, INVALID_GROUP };
 
-	if (v == nullptr || (!Group::IsValidID(new_g) && !IsDefaultGroupID(new_g) && new_g != NEW_GROUP)) return { CMD_ERROR, INVALID_GROUP };
+	VehicleList list;
+	if (veh_id == INVALID_VEHICLE && vli.Valid()) {
+		if (!GenerateVehicleSortList(&list, vli) || list.empty()) return { CMD_ERROR, INVALID_GROUP };
+	} else {
+		Vehicle *v = Vehicle::GetIfValid(veh_id);
+		if (v == nullptr) return { CMD_ERROR, INVALID_GROUP };
+		list.push_back(v);
+	}
+
+	VehicleType vtype = list.front()->type;
+	for (const Vehicle *v : list) {
+		if (v->owner != _current_company || !v->IsPrimaryVehicle()) return { CMD_ERROR, INVALID_GROUP };
```

Comment:
Most of the checks in this function are just to prevent bad actors (assume all information is invalid), or issues when the state changes in between, right?
However, I think I can, as user, trigger this error when I get the list of vehicles of an oil rig. If you're the only company it works fine, but when another company joins you're in trouble. It might be hard to see that there is a vehicle from another company in there, so would it be useful to have a specific error message in case the owner is not the current company hinting about the fact that there are vehicles from another company in the list?

Or should it maybe just remove the vehicle of other companies from the list, so it's easier to work with for shared stations?

Reply:
Yeah, ignoring wrong owners seems like the best way.

Reply:
Well, turns out there is no "Manage Vehicles" menu on a shared station...